### PR TITLE
Fix NullPointerException in cquery --output=graph.

### PR DESCRIPTION
**Pull request by gregestren**
Fixes https://github.com/bazelbuild/bazel/issues/12915.

The repro from the bug crashes because these two deps are compared:

`Target: @remote_java_tools_linux//:java_tools/src/tools/singlejar/singlejar_local`
`Config: d8d0eb07eb92791668ac973282be1523379b0025b22f0ade56b13d519f2feb2a`

`Target: @remote_java_tools_linux//:java_tools/src/tools/singlejar/singlejar_local`
`Config: null`

This is a `cc_binary`: https://github.com/bazelbuild/bazel/blob/0c1257ed4e1b83f8d0f6c79d641f6bfcf4d1cfc4/src/tools/singlejar/BUILD#L92-L93

I have no idea why the graph visitor tags one of its instances with a null configuration. Nevertheless, this code is still safer.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/query2/cquery/GraphOutputFormatterCallback.java [hunk 1]
```diff
@@ -50,9 +51,20 @@ public String getName() {
...
               Label label2 = ct2.getLabel();
-              return label1.equals(label2)
-                  ? ct1.getConfigurationChecksum().compareTo(ct2.getConfigurationChecksum())
-                  : label1.compareTo(label2);
+              if (!label1.equals(label2)) {
+                return label1.compareTo(label2);
+              }
+              String checksum1 = ct1.getConfigurationChecksum();
+              String checksum2 = ct2.getConfigurationChecksum();
+              if (Objects.equals(checksum1, checksum2)) {
```
Comment (katre) [inline src/main/java/com/google/devtools/build/lib/query2/cquery/GraphOutputFormatterCallback.java:59]: This is the same as the compareTo below, and the checksum2 null check isn't needed, either. I'd simplify this as:

if (checksum1 == null) {
  return -1;
} else {
  return checksum1.compareTo(checksum2);
}

**Review by katre — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/devtools/build/lib/query2/cquery/GraphOutputFormatterCallback.java (hunk 1 above)
Comment (gregestren) [reply src/main/java/com/google/devtools/build/lib/query2/cquery/GraphOutputFormatterCallback.java]: How's this? I left the double null check because of this comment from https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html:

> Note that null is not an instance of any class, and e.compareTo(null) should throw a NullPointerException even though e.equals(null) returns false.

Comment (gregestren) [reply src/main/java/com/google/devtools/build/lib/query2/cquery/GraphOutputFormatterCallback.java]: Or am I missing something about the null check?

**Comment by gregestren** [comment]
Also note I don't know how to add a test since I don't know how to invoke a dep simultaneously in a null and non-null configuration.

It's probably worth exploring how the query logic decides a `cc_binary` appears in both configurations. Not sure I should worry about that for the scope of this PR (scope creep). But open to suggestions.

File: src/main/java/com/google/devtools/build/lib/query2/cquery/GraphOutputFormatterCallback.java (hunk 1 above)
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/query2/cquery/GraphOutputFormatterCallback.java]: Ah, I hadn't realized that. Okay, the second null check is necessary, then.

**Review by katre — APPROVED** [review]
